File: algorithms/cpp/338.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> bits(num+1, 0);
        for (int i = 1; i <= num; i++)
            bits[i] = bits[i>>1] + (i&1);
        return bits;
    }
};

int main() {
    return 0;
}

File: algorithms/python/338.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        bits = [0] * (num+1)
        for i in range(1, num+1):
            bits[i] = bits[i>>1] + (i&1)
        return bits
